### PR TITLE
TUP-26609 - Harden deserialization for codegen

### PR DESCRIPTION
**Pull request by coheigea**
This task is to harden deserialization in tdi-studio-se/main/plugins/org.talend.designer.codegen, to make sure that we are only deserializing what we are expecting.

Note: I can't build the code, so it may have compilation errors - it needs to be tested also.

### CODE REVIEW

File: main/plugins/org.talend.designer.codegen/src/main/java/org/talend/designer/codegen/model/JetSkeletonManager.java [hunk 1]
```diff
@@ -170,7 +172,18 @@ private void deserializeAlreadyChecked() throws Exception {
         BufferedInputStream bufferedInputStream = null;
         try {
             bufferedInputStream = new BufferedInputStream(new FileInputStream(file));
-            ObjectInputStream objectIn = new ObjectInputStream(bufferedInputStream);
+            ObjectInputStream objectIn = new ObjectInputStream(bufferedInputStream) {
+                @Override
+                protected Class<?> resolveClass(ObjectStreamClass desc) throws IOException, ClassNotFoundException {
+                    // Enforce the types that we are about to deserialize
+                    if (!(desc.getName().startsWith("java.util")
```
Comment (wwang-talend) [inline main/plugins/org.talend.designer.codegen/src/main/java/org/talend/designer/codegen/model/JetSkeletonManager.java:179]: the package "java.util.iambadguy" is possible?

Comment (wwang-talend) [reply main/plugins/org.talend.designer.codegen/src/main/java/org/talend/designer/codegen/model/JetSkeletonManager.java:179]: then can skip the security check

**Review by wwang-talend — APPROVED** [review]

File: main/plugins/org.talend.designer.codegen/src/main/java/org/talend/designer/codegen/model/JetSkeletonManager.java (hunk 1 above)
Comment (coheigea) [reply main/plugins/org.talend.designer.codegen/src/main/java/org/talend/designer/codegen/model/JetSkeletonManager.java:179]: In theory yes, but remember the package has to be on the classpath of the application, so it's unlikely that the application will be allowing this package ;-)

**Review by wwang-talend — APPROVED** [review]

**Comment by nrousseau** [comment]
seems code was re-written... not sure why... closing the PR since it's done

**Pull request closed without merging**